### ISSUE TITLE
Bug(`sort-imports`): False positive error report with `sortSideEffects: false`

### ISSUE THREAD

**Issue by hugop95** [azat-io/eslint-plugin-perfectionist#315]
### Describe the bug

The code below reports an error that should not exist. Auto-fixing does nothing. 

### Code example

Configuration:

```
{
  "sortSideEffects": true,
  "newlinesBetween": "never",
  "groups": [
    "unknown"
  ]
}
```

Code:
```ts
import a from "./a";
import b from "./b";
import './a-side-effect'
```

Error: `Expected "./a-side-effect" to come before "./b"  perfectionist/sort-imports`

### ESLint version

8.57.0

### ESLint Plugin Perfectionist version

3.8.0

### Validations

- [X] Read the [docs](https://perfectionist.dev).
- [X] Check that there isn't [already an issue](https://github.com/azat-io/eslint-plugin-perfectionist/issues) that reports the same bug to avoid creating a duplicate.
